distribution/push_v2.go

```diff
@@ -50,14 +49,14 @@ func (p *v2Pusher) Push() (fallback bool, err error) {
 
 	localName := p.repoInfo.LocalName.Name()
 
-	var associations []tag.Association
-	if _, isTagged := p.ref.(reference.Tagged); isTagged {
-		imageID, err := p.config.TagStore.Get(p.ref)
+	var associations []reference.Association
+	if _, isTagged := p.ref.(reference.NamedTagged); isTagged {
+		imageID, err := p.config.ReferenceStore.Get(p.ref)
 		if err != nil {
 			return false, fmt.Errorf("tag does not exist: %s", p.ref.String())
 		}
 
-		associations = []tag.Association{
+		associations = []reference.Association{
```

Comment:
I like how `Association` is in the same package as the other reference types now.